prow/cmd/peribolos/main.go

```diff
@@ -255,23 +286,155 @@ func configureOrg(opt options, client *github.Client, orgName string, orgConfig
 
 	// teams = set(all teams)
 	for name, team := range orgConfig.Teams {
-		team.Name = name
-		num := -1
-		// find team name in teams (or previous name)
-		if err := configureTeam(client, num, team); err != nil {
+		_ = name
+		id, err := configureTeam(client, orgName, team)
+		if err != nil {
 			return fmt.Errorf("failed to update %s: %v", orgName, err)
 		}
+		return configureTeamMembers(client, *id, team)
 	}
 	return nil
 }
 
-func configureTeam(client *github.Client, num int, team org.Team) error {
-	// create team if num < 0
-	// else diff and patch
+type findTeamClient interface {
+	ListTeams(org string) ([]github.Team, error)
+}
+
+type configureTeamClient interface {
+	findTeamClient
+	CreateTeam(org, name string) (*github.Team, error)
+	EditTeam(id int, team github.Team) (*github.Team, error)
+}
+
+func findTeam(client findTeamClient, org, name string, previousNames ...string) (*github.Team, error) {
+	teams, err := client.ListTeams(org)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list %s teams: %v", org, err)
+	}
+	// Does a team exactly match the current name?
+	for _, t := range teams {
```

Comment:
Because we aren't tracking teams by the same unique ID that github does (we're looking for name, while they are looking for id number), should we have some config validation around uniqueness in the config file? For example, if a team name is listed twice in the config, or if a team name is listed as a defined team, but also a "previous" team?

Also, what happens to teams that aren't defined? Right now, it looks like nothing.

Reply:
+1